stdlib/toolchain/CMakeLists.txt

```diff
@@ -48,8 +48,17 @@ list(APPEND CXX_COMPILE_FLAGS "-DSWIFT_COMPATIBILITY_LIBRARY=1")
 
 if(SWIFT_STDLIB_SUPPORT_BACK_DEPLOYMENT)
   add_subdirectory(legacy_layouts)
-  add_subdirectory(Compatibility50)
-  add_subdirectory(Compatibility51)
-  add_subdirectory(CompatibilityDynamicReplacements)
-  add_subdirectory(CompatibilityConcurrency)
+  foreach(compatibility_name IN LISTS COMPATIBILITY_NAMES)
+    add_subdirectory("Compatibility${compatibility_name}")
+  endforeach()
+
+  set(vsuffix "-${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}-${SWIFT_HOST_VARIANT_ARCH}")
```

Comment:
I preferred to keep the suffix here so target and architecture are explicit in the name of this target